org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java

```diff
@@ -14,72 +14,22 @@
 
 import java.nio.file.Path;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Stream;
 
 import org.eclipse.aether.graph.DependencyNode;
 
 import aQute.bnd.osgi.Jar;
 
-public final class WrappedBundle {
-
-	private DependencyNode node;
-	private List<WrappedBundle> depends;
-	private String instructionsKey;
-	private Path file;
-	private Jar jar;
-	private List<ProcessingMessage> messages;
-
-	WrappedBundle(DependencyNode node, List<WrappedBundle> depends, String key, Path file, Jar jar,
-			List<ProcessingMessage> messages) {
-		this.node = node;
-		this.depends = depends;
-		this.instructionsKey = key;
-		this.file = file;
-		this.jar = jar;
-		this.messages = messages;
-	}
-
-	String getInstructionsKey() {
-		return instructionsKey;
-	}
-
-	Jar getJar() {
-		return jar;
-	}
-
-	/**
-	 * @return the file where the wrappes bundle is located
-	 */
-	public Path getFile() {
-		return file;
-	}
+public record WrappedBundle(DependencyNode node, List<WrappedBundle> depends, String instructionsKey,
+		/** the location of the wrapped bundle's files */
+		Path file, //
+		Jar jar, List<ProcessingMessage> ownMessages) {
 
 	/**
 	 * @return the messages that where produced
 	 */
 	public Stream<ProcessingMessage> messages() {
-		return Stream.concat(messages.stream(), depends.stream().flatMap(dep -> dep.messages()));
-	}
-
-	@Override
-	public int hashCode() {
-		return Objects.hash(instructionsKey, node);
```

Comment:
As far as I can tell this class (now record) is not used as key in any map.
Therefore I think it is not necessary to override hashCode/equals.

Reply:
The idea was/is that code can use this to decide if two items are equal without having to know the internal state, even though it is currently not used (at m2e) in such a way.

Reply:
Acknowledged.